### PR TITLE
[bitnami/redis] Add new environment variable REDIS_TLS_CA_DIR to support tls-ca-cert-dir config in redis.conf

### LINKED ISSUE #56361: [bitnami/redis] Support SSL certificates from a public CA by making reading ca-certificates from os configurable
### Name and Version

bitnami/redis

### What is the problem this feature will solve?

redis supports these configurations for CA certificates in redis.conf
```
# tls-ca-cert-file ca.crt
# tls-ca-cert-dir /etc/ssl/certs
```
currently there is an environment variable for `tls-ca-cert-file` (REDIS_TLS_CA_FILE) but not `tls-ca-cert-dir`
https://redis.io/docs/management/config-file/
if `tls-ca-cert-dir` can be set as an environment variable, it can be pointed to the OS directory that holds ca-certificates
eg  -` /usr/share/ca-certificates/mozilla/`
and public certificates like letsencrypt can be used for redis when `tls-auth-clients no`

### What is the feature you are proposing to solve the problem?

add a new environment variable REDIS_TLS_CA_DIR that is the equivalent of `tls-ca-cert-dir` in redis.conf

### What alternatives have you considered?

It is possible to supply a custom config to redis and set this directly in redis.conf

### PR DESCRIPTION
**Pull request by nebed**
Add new environment variable REDIS_TLS_CA_DIR to [bitnami/redis] in order to support tls-ca-cert-dir config in redis.conf

### Description of the change

Added an additional environment variable that can be configured when enabling TLS for redis - `REDIS_TLS_CA_DIR`. 
This environment variable translates to the `tls-ca-cert-dir` config in `redis.conf`. and let's you supply a directory with several CA certificates instead of a single CA file, as in the case of os provided CA certificates.

The `REDIS_TLS_CA_FILE` environment variable has precedence over the `REDIS_TLS_CA_DIR` environment variable, when they are both specified, only `REDIS_TLS_CA_FILE` will be used

### Benefits

This will make it easier to use SSL certificates generated by public CAs as TLS for redis containers as you can point the directory to the os provided ca certificates.
Also this just makes it easier to set a config already supported in redis.conf

### Applicable issues

- fixes #56361

### CODE REVIEW

File: bitnami/redis/6.2/debian-11/rootfs/opt/bitnami/scripts/libredis.sh [hunk 1]
```diff
@@ -265,7 +269,9 @@ redis_configure_replication() {
     elif [[ "$REDIS_REPLICATION_MODE" =~ ^(slave|replica)$ ]]; then
         if [[ -n "$REDIS_SENTINEL_HOST" ]]; then
             local -a sentinel_info_command=("redis-cli" "-h" "${REDIS_SENTINEL_HOST}" "-p" "${REDIS_SENTINEL_PORT_NUMBER}")
-            is_boolean_yes "$REDIS_TLS_ENABLED" && sentinel_info_command+=("--tls" "--cert" "${REDIS_TLS_CERT_FILE}" "--key" "${REDIS_TLS_KEY_FILE}" "--cacert" "${REDIS_TLS_CA_FILE}")
+            is_boolean_yes "$REDIS_TLS_ENABLED" && sentinel_info_command+=("--tls" "--cert" "${REDIS_TLS_CERT_FILE}" "--key" "${REDIS_TLS_KEY_FILE}")
+            is_empty_value "$REDIS_TLS_CA_FILE" && sentinel_info_command+=("--cacertdir" "${REDIS_TLS_CA_DIR}")
+            ! is_empty_value "$REDIS_TLS_CA_FILE" && sentinel_info_command+=("--cacert" "${REDIS_TLS_CA_FILE}")
```
Comment (juan131) [inline bitnami/redis/6.2/debian-11/rootfs/opt/bitnami/scripts/libredis.sh:274]: You can do this in a single line:
```suggestion
is_empty_value "$REDIS_TLS_CA_FILE" && sentinel_info_command+=("--cacertdir" "${REDIS_TLS_CA_DIR}") || sentinel_info_command+=("--cacert" "${REDIS_TLS_CA_FILE}")
```

File: bitnami/redis/6.2/debian-11/rootfs/opt/bitnami/scripts/libredis.sh [hunk 2]
```diff
@@ -431,7 +437,8 @@ redis_configure_default() {
             fi
             redis_conf_set tls-cert-file "$REDIS_TLS_CERT_FILE"
             redis_conf_set tls-key-file "$REDIS_TLS_KEY_FILE"
-            redis_conf_set tls-ca-cert-file "$REDIS_TLS_CA_FILE"
+            ! is_empty_value "$REDIS_TLS_CA_FILE" && redis_conf_set tls-ca-cert-file "$REDIS_TLS_CA_FILE"
+            is_empty_value "$REDIS_TLS_CA_FILE" && redis_conf_set tls-ca-cert-dir "$REDIS_TLS_CA_DIR"
```
Comment (juan131) [inline bitnami/redis/6.2/debian-11/rootfs/opt/bitnami/scripts/libredis.sh:441]: Same here:
```suggestion
is_empty_value "$REDIS_TLS_CA_FILE" && redis_conf_set tls-ca-cert-dir "$REDIS_TLS_CA_DIR" || redis_conf_set tls-ca-cert-file "$REDIS_TLS_CA_FILE"
```

File: bitnami/redis/README.md [hunk 3]
```diff
@@ -476,9 +476,12 @@ Starting with version 6, Redis(R) adds the support for SSL/TLS connections. Shou
 * `REDIS_TLS_CERT_FILE`: File containing the certificate file for the TLS traffic. No defaults.
 * `REDIS_TLS_KEY_FILE`: File containing the key for certificate. No defaults.
 * `REDIS_TLS_CA_FILE`: File containing the CA of the certificate. No defaults.
+* `REDIS_TLS_CA_DIR`: Directory containing the CA certificates. No defaults.
 * `REDIS_TLS_DH_PARAMS_FILE`: File containing DH params (in order to support DH based ciphers). No defaults.
 * `REDIS_TLS_AUTH_CLIENTS`: Whether to require clients to authenticate or not. Defaults to `yes`.
 
+When both `REDIS_TLS_CA_FILE` and `REDIS_TLS_CA_DIR` are set, `REDIS_TLS_CA_FILE` is used preferentially.
+
```
Comment (juan131) [inline bitnami/redis/README.md:484]:
```suggestion
* `REDIS_TLS_CA_FILE`: File containing the CA of the certificate (takes precedence over `REDIS_TLS_CA_DIR`). No defaults.
* `REDIS_TLS_CA_DIR`: Directory containing the CA certificates. No defaults.
* `REDIS_TLS_DH_PARAMS_FILE`: File containing DH params (in order to support DH based ciphers). No defaults.
* `REDIS_TLS_AUTH_CLIENTS`: Whether to require clients to authenticate or not. Defaults to `yes`.
```

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks so much for this awesome contribution and sorry for the delay in the review @nebed ! Please check my comments when you have a chance.

**Comment by nebed** [comment]
Thanks for the suggestions @juan131, i have replicated them in the other containers

**Review by juan131 — APPROVED** [review]
Awesome, LGTM! New container versions (for each branch) will be released within the next 24 hrs including your changes, thanks again for your contribution.

**Pull request merged**